### PR TITLE
Add node8 firebase snippets

### PR DESCRIPTION
**Pull request by ace-n**
cc @stew-r @samtstern

### CODE REVIEW

File: functions/node8/index.js [hunk 1]
```diff
@@ -123,3 +123,61 @@ exports.helloGCSGeneric = (data, context) => {
...
+  console.log(JSON.stringify(data.oldValue, null, 2));
+
+  console.log(`\nNew value:`);
+  console.log(JSON.stringify(data.value, null, 2));
+};
+// [END functions_firebase_firestore_node8]
+
+// [START functions_firebase_auth_node8]
+/**
+ * Triggered by a change to a Firebase Auth user object.
```
Comment (samtstern) [inline functions/node8/index.js]: Same comment as before: it's only creation or deletion, not a change.

Comment (ace-n) [reply functions/node8/index.js]: Done.

File: functions/node8/index.js [hunk 2]
```diff
@@ -123,3 +123,61 @@ exports.helloGCSGeneric = (data, context) => {
...
+ * Triggered by a change to a Firebase RTDB reference.
+ *
+ * @param {object} data The event payload.
+ * @param {object} context The event metadata.
+ */
+exports.helloRTDB = (data, context) => {
+  const triggerResource = context.resource;
+
+  console.log(`Function triggered by change to: ${triggerResource}`);
+  console.log(`Admin?: ${!!data.admin}`);
```
Comment (ace-n) [inline functions/node8/index.js]: @stew-r is this correct?

**Comment by ace-n** [comment]
TODO: add tests

**Comment by ace-n** [comment]
Tests have been added - @fhinkel please review?

File: functions/node8/index.js [hunk 3]
```diff
@@ -123,3 +123,61 @@ exports.helloGCSGeneric = (data, context) => {
...
+ * @param {object} data The event payload.
+ * @param {object} context The event metadata.
+ */
+exports.helloFirestore = (data, context) => {
+  const triggerResource = context.resource;
+
+  console.log(`Function triggered by change to: ${triggerResource}`);
+
+  console.log(`\nOld value:`);
+  console.log(JSON.stringify(data.oldValue, null, 2));
```
Comment (fhinkel) [inline functions/node8/index.js]: Is the double space necessary? I personally would prefer shorter code without the extra params.

File: functions/node8/package.json [hunk 4]
```diff
@@ -19,7 +19,9 @@
     "url": "https://github.com/GoogleCloudPlatform/nodejs-docs-samples/issues"
   },
   "dependencies": {
-    "ava": "^0.25.0"
+    "@google-cloud/nodejs-repo-tools": "^2.3.3",
```
Comment (fhinkel) [inline functions/node8/package.json]: `repo-tools` and `ava` should be dev dependencies.

File: functions/node8/test/index.test.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const uuid = require('uuid');
+const test = require(`ava`);
```
Comment (fhinkel) [inline functions/node8/test/index.test.js]: Inconsistent quotes.

File: functions/node8/index.js (hunk 3 above)
Comment (ace-n) [reply functions/node8/index.js]: Not _strictly_ necessary, but makes it easier to read. (Without it, everything is compressed into one line.)

File: functions/node8/package.json (hunk 4 above)
Comment (ace-n) [reply functions/node8/package.json]: Done.

**Pull request merged**